conf/config.php

```diff
@@ -1,47 +1,36 @@
 <?php
+
 	// *******************************************
 	// *** Database configuration (important!) ***
 	// *******************************************
 
-	define('DB_TYPE', "mysql"); // or mysql
-	define('DB_HOST', "localhost");
-	define('DB_USER', "__DB_NAME__");
-	define('DB_NAME', "__DB_NAME__");
-	define('DB_PASS', "__DB_PWD__");
-	define('DB_PORT', '3306'); // usually 5432 for PostgreSQL, 3306 for MySQL
+	putenv('TTRSS_DB_TYPE=mysql'); // or mysql
+	putenv('TTRSS_DB_HOST=localhost');
+	putenv('TTRSS_DB_USER=__DB_NAME__');
+	putenv('TTRSS_DB_NAME=__DB_NAME__');
+	putenv('TTRSS_DB_PASS=__DB_PWD__');
+	putenv('TTRSS_DB_PORT=3306'); // usually 5432 for PostgreSQL, 3306 for MySQL
 
-	define('MYSQL_CHARSET', 'UTF8');
+	putenv('TTRSS_MYSQL_CHARSET=UTF8');
 	// Connection charset for MySQL. If you have a legacy database and/or experience
 	// garbage unicode characters with this option, try setting it to a blank string.
 
 	// ***********************************
 	// *** Basic settings (important!) ***
 	// ***********************************
 
-	define('SELF_URL_PATH', '__DOMAIN_PATH__');
+	putenv('TTRSS_SELF_URL_PATH=__DOMAIN_PATH__');
 	// Full URL of your tt-rss installation. This should be set to the
 	// location of tt-rss directory, e.g. http://example.org/tt-rss/
 	// You need to set this option correctly otherwise several features
 	// including PUSH, bookmarklets and browser integration will not work properly.
 
-	define('FEED_CRYPT_KEY', '');
-	// WARNING: mcrypt is deprecated in php 7.1. This directive exists for backwards
-	// compatibility with existing installs, new passwords are NOT going to be encrypted.
-	// Use update.php --decrypt-feeds to decrypt existing passwords in the database while
-	// mcrypt is still available.
-
-	// Key used for encryption of passwords for password-protected feeds
-	// in the database. A string of 24 random characters. If left blank, encryption
-	// is not used. Requires mcrypt functions.
-	// Warning: changing this key will make your stored feed passwords impossible
-	// to decrypt.
-	
-	define('SINGLE_USER_MODE', false);
+	putenv('TTRSS_SINGLE_USER_MODE=true');
```

Comment:
Seems very strange to activate that ?
What happens for instances with multiple user ?

Reply:
Patches that allows to connect to the app without authentication are not working anymore.
I see 3 possible solutions:

-  rework patches to work with upstream changes
-  use built-in authentication
-  set authentication to false and protect app access with SSO (as in this PR)

I am unable to fix the current patchs to the upstream changes.